src/Database/LSMTree/Monoidal.hs

```diff
@@ -148,8 +148,8 @@ data Range k
 
 -- | Perform a batch of lookups.
 --
--- Lookups can be issued concurrently from multiple Haskell threads, and/or in
--- parallel from multiple OS threads.
+-- Lookups can be issued concurrently and in parallel from multiple Haskell
+-- threads.
```

Comment:
I think it's even simpler:
```
Lookups can be performed concurrently from multiple Haskell threads.
```
That's it. We don't need to say parallel. It's an implementation detail that GHC threads can be parallel. The important point is they're concurrent, and we're saying it's safe to use concurrently.